Tests/MongoSwiftSyncTests/UnifiedTestRunner/EntityDescription.swift

```diff
@@ -0,0 +1,117 @@
+import Foundation
+import MongoSwiftSync
+
+/// Represents a description of an entity (client, db, etc.).
+enum EntityDescription: Decodable {
+    case client(Client)
+    case database(Database)
+    case collection(Coll)
+    case session(Session)
+    case bucket(Bucket)
+
+    /// Describes a Client entity.
+    struct Client: Decodable {
+        /// Unique name for this entity.
+        let id: String
+
+        /// Additional URI options to apply to the test suite's connection string that is used to create this client.
+        let uriOptions: MongoClientOptions?
+
+        /**
+         * If true and the topology is a sharded cluster, the test runner MUST assert that this MongoClient connects
+         * to multiple mongos hosts (e.g. by inspecting the connection string). If false and the topology is a sharded
+         * cluster, the test runner MUST ensure that this MongoClient connects to only a single mongos host (e.g. by
+         * modifying the connection string).
+         * If this option is not specified and the topology is a sharded cluster, the test runner MUST NOT enforce any
+         * limit on the number of mongos hosts in the connection string. This option has no effect for non-sharded
+         * topologies.
+         */
+        let useMultipleMongoses: Bool?
+
+        /// Types of events that can be observed for this client. Unspecified event types MUST be ignored by this
+        /// client's event listeners.
+        let observeEvents: [String]?
+
+        /// Command names for which the test runner MUST ignore any observed command monitoring events. The command(s)
+        /// will be ignored in addition to configureFailPoint and any commands containing sensitive information (per
+        /// the Command Monitoring spec).
+        let ignoreCommandMonitoringEvents: [String]?
+    }
+
+    /// Describes a Database entity.
+    struct Database: Decodable {
+        /// Unique name for this entity.
+        let id: String
+
+        /// Client entity from which this database will be created.
+        let client: String
+
+        /// Database name.
+        let databaseName: String
+
+        /// Options to use for this database.
+        let options: MongoDatabaseOptions?
+    }
+
+    /// Describes a Collection entity. (named Coll because Collection clashes with the stdlib's Collection protocol.)
+    struct Coll: Decodable {
+        /// Unique name for this entity.
+        let id: String
+
+        /// Database entity from which this collection will be created.
+        let database: String
+
+        /// Collection name.
+        let collectionName: String
+
+        /// Options to use for this collection.
+        let options: MongoCollectionOptions?
```

Comment:
according to the [schema](https://github.com/mongodb/specifications/blob/master/source/unified-test-format/schema-1.0.json#L105), these options types are prefixed with the name of the entity theyre configuring. e.g. this one would be "collectionOptions". Ditto for session, database, bucket, etc.

Reply:
🤦‍♀️  great catch, fixed.